android/engine/src/main/java/org/smartregister/fhircore/engine/task/FhirCarePlanGenerator.kt

```diff
@@ -192,8 +192,10 @@ constructor(
       if (carePlan.status == CarePlan.CarePlanStatus.COMPLETED)
         carePlan
           .activity
+          .asSequence()
           .flatMap { it.outcomeReference }
           .filter { it.reference.startsWith(ResourceType.Task.name) }
+          .toList()
```

Comment:
Why are we introducing this lambda here? The return type for a filter operation is already a list. This `toList` is redundant and will impact the performance of the operations

Reply:
@ellykits we can have a discussion about this.